plugins/woocommerce-blocks/tests/e2e/tests/product-gallery/product-gallery.block_theme.spec.ts

```diff
@@ -326,15 +367,9 @@ test.describe( `${ blockData.name }`, () => {
 			} );
 			await largeImageBlock.click();
 
-			const productGalleryPopUpContent = page.locator(
-				'.wc-block-product-gallery-dialog__body'
-			);
-
-			const popUpSelectedImageId = await getVisibleLargeImageId(
-				productGalleryPopUpContent.locator(
-					`[data-block-name="woocommerce/product-gallery-large-image"]`
-				)
-			);
+			// eslint-disable-next-line playwright/no-wait-for-timeout, no-restricted-syntax
+			await page.waitForTimeout( 300 );
```

Comment:
I need to somehow avoid this. The test evaluates before the side effect runs which focuses on the image element in `dialogStateChange`

Reply:
I see, we're evaluating JS so yeah, it doesn't wait for stuff. I think we could await for dialog to be visible before running `getActiveElementImageId`. Did you try it?